packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -0,0 +1,399 @@
+
+    destinationInfoPlist.writeAsBytesSync(sourceInfoPlist.readAsBytesSync());
+
+    final Status status = logger.startProgress(' ├─Assembling Flutter resources for App.framework...', timeout: timeoutConfiguration.slowOperation);
+    await bundleBuilder.build(
+      platform: TargetPlatform.ios,
+      buildMode: mode,
+      // Relative paths show noise in the compiler https://github.com/dart-lang/sdk/issues/37978.
+      mainPath: fs.path.absolute(targetFile),
+      assetDirPath: destinationAppFrameworkDirectory.childDirectory('flutter_assets').path,
+      precompiledSnapshot: mode != BuildMode.debug,
+    );
+    status.stop();
+  }
+
+  Future<void> _produceAotAppFrameworkIfNeeded(BuildMode mode, Directory iPhoneBuildOutput, Directory destinationAppFrameworkDirectory) async {
+    if (mode == BuildMode.debug) {
+      return;
+    }
+    final Status status = logger.startProgress(' ├─Building Dart AOT for App.framework...', timeout: timeoutConfiguration.slowOperation);
+    await aotBuilder.build(
+      platform: TargetPlatform.ios,
+      outputPath: iPhoneBuildOutput.path,
+      buildMode: mode,
+      // Relative paths show noise in the compiler https://github.com/dart-lang/sdk/issues/37978.
+      mainDartFile: fs.path.absolute(targetFile),
+      quiet: true,
+      reportTimings: false,
+      iosBuildArchs: <DarwinArch>[DarwinArch.armv7, DarwinArch.arm64],
+    );
+
+    const String appFrameworkName = 'App.framework';
+    copyDirectorySync(iPhoneBuildOutput.childDirectory(appFrameworkName), destinationAppFrameworkDirectory);
+    status.stop();
+  }
+
+  Future<void> _producePlugins(
+    FlutterProject flutterProject,
+    String xcodeBuildConfiguration,
+    Directory iPhoneBuildOutput,
+    Directory simulatorBuildOutput,
+    Directory modeDirectory,
+    Directory outputDirectory,
+  ) async {
+    final Status status = logger.startProgress(' ├─Building plugins...', timeout: timeoutConfiguration.slowOperation);
+    List<String> pluginsBuildCommand = <String>[
+      'xcrun',
+      'xcodebuild',
+      '-alltargets',
+      '-sdk',
+      'iphoneos',
+      '-configuration',
+      xcodeBuildConfiguration,
+      'SYMROOT=${iPhoneBuildOutput.path}',
+      'ONLY_ACTIVE_ARCH=NO' // No device targeted, so build all valid architectures.
+    ];
+
+    final FlutterProject flutterProject = FlutterProject.current();
+    await processUtils.run(
+      pluginsBuildCommand,
+      workingDirectory: flutterProject.ios.hostAppRoot.childDirectory('Pods').path,
+      allowReentrantFlutter: false,
+    );
+
+    pluginsBuildCommand = <String>[
+      'xcrun',
+      'xcodebuild',
+      '-alltargets',
+      '-sdk',
+      'iphonesimulator',
+      '-configuration',
+      xcodeBuildConfiguration,
+      'SYMROOT=${simulatorBuildOutput.path}',
+      'ARCHS=x86_64',
+      'ONLY_ACTIVE_ARCH=NO' // No device targeted, so build all valid architectures.
```

Comment:
I don't know if `ONLY_ACTIVE_ARCH` is necessary since `ARCHS` is specified (again, I know I wrote it, my bad...)

Reply:
I don't know if it has a default value but I think it can blow up with a 'no architecture to compile for' if it has an active/connected 32-bit simulator (on older macOSes?) and wants x64.